File: src/main/java/com/matag/game/turn/action/finish/FinishGameService.java
package com.matag.game.turn.action.finish;

import org.springframework.stereotype.Component;

import com.matag.game.adminclient.AdminClient;
import com.matag.game.player.Player;
import com.matag.game.status.GameStatus;

import lombok.AllArgsConstructor;

@Component
@AllArgsConstructor
public class FinishGameService {
  private final AdminClient adminClient;

  public void setWinner(GameStatus gameStatus, Player player) {
    gameStatus.getTurn().setWinner(player.getName());
    adminClient.finishGame(gameStatus.getGameId(), player);
  }
}

File: src/main/kotlin/com/matag/game/turn/action/finish/FinishGameService.kt
package com.matag.game.turn.action.finish

import com.matag.game.adminclient.AdminClient
import com.matag.game.player.Player
import com.matag.game.status.GameStatus
import org.springframework.stereotype.Component

@Component
class FinishGameService(
    private val adminClient: AdminClient
) {
    fun setWinner(gameStatus: GameStatus, player: Player) {
        gameStatus.turn.winner = player.name
        adminClient.finishGame(gameStatus.gameId, player)
    }
}
